DEPRECATIONS.rst

```diff
@@ -0,0 +1,101 @@
+Deprecations: how to migrate
+============================
+
+New Tracer interface
+--------------------
+
+The following classes and methods are deprecated and should not be used.
+Please use the Thread interface instead.
+They will be removed in version 1.0.
+
+REASON: they don't work well in multi-threaded environments and can cause
+dropped spans or memory leaks.
+
+- py_zipkin.storage.Stack
+- py_zipkin.storage.ThreadLocalStack
+- py_zipkin.storage.SpanStorage
+- py_zipkin.storage.default_span_storage
+- py_zipkin.thread_local.get_thread_local_zipkin_attrs
+- py_zipkin.thread_local.get_thread_local_span_storage
+- py_zipkin.thread_local.get_zipkin_attrs
+- py_zipkin.thread_local.pop_zipkin_attrs
+- py_zipkin.thread_local.push_zipkin_attrs
+
+To access the current zipkin_attrs
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code-block:: python
+
+    from py_zipkin import get_default_tracer
+
+    zipkin_attrs = get_default_tracer().get_zipkin_attrs()
+    get_default_tracer().push_zipkin_attrs(zipkin_attrs)
+
+To override the default tracer and provide your own in a multi-threaded env
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can instantiate your own Tracer instance and pass it to ALL your zipkin_span
+context managers and decorators. You'll need to propagate the tracer around on
+your own.
+
+.. code-block:: python
+
+    from py_zipkin import Tracer
+
+    tracer = Tracer()
+    with zipkin_span(
+        service_name="homepage",
+        span_name="get /home",
+        transport=MockTransport(),
+        sample_rate=100.0,
+        get_tracer=lambda: tracer,
+    ):
+        # do stuff
+
+Alternatively you can use the `get_default_tracer` and `set_default_tracer`
+functions to reset the default tracer every time you switch thread or
```

Comment:
we use the term "current span" and "current tracer" in brave in case that is interesting to you

Reply:
Yeah I thought about it and decided against using "current". The issue is that since we support passing in a custom tracer to every context manager there's no way for a generic helper like this to return that. So I think "default" is a better name since it users are not passing in a custom one "default" sound what you're looking for, while if you use a custom tracer then you'll realize this is a different one

Reply:
that's fine was just mentioning in case it hadn't been considered.thx